File: src/main/java/com/waitring/waitring/entity/Menu.java
package com.waitring.waitring.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;

/**
 * 메뉴 엔티티
 */

@Entity
@Getter
@ToString(exclude = "store")
@Builder
@AllArgsConstructor
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Menu extends BaseEntity {

    @Id @GeneratedValue
    private Long id;

    private String name; // 메뉴명
    private Integer price; // 메뉴 가격
    private String detail; // 메뉴 설명
    @Column(columnDefinition = "TEXT")
    private String image; // 메뉴 이미지

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "store_id")
    @JsonIgnore
    private Store store; // 메뉴가 속한 가게
}

File: src/main/java/com/waitring/waitring/entity/Menu.kt
package com.waitring.waitring.entity

import javax.persistence.*

/**
 * 메뉴 엔티티
 */
@Entity
class Menu(name: String, price: Int, store: Store, image: String? = null, detail: String? = "") : BaseEntity() {
    @Id
    @GeneratedValue
    var id: Long? = null

    @Column(nullable = false)
    var name = name // 메뉴명
        protected set

    @Column(nullable = false)
    var price = price // 메뉴 가격
        protected set

    var detail = detail // 메뉴 설명
        protected set

    @Column(columnDefinition = "TEXT")
    var image = image // 메뉴 이미지
        protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "store_id", nullable = false, updatable = false)
    var store = store // 메뉴가 속한 가게
        protected set

    override fun toString(): String {
        return "Menu(id=$id, name=$name, price=$price, detail=$detail, image=$image, store=$store)"
    }
}
